src/dfa/CM_Global.h

```diff
@@ -118,40 +118,14 @@ enum options
 #define QUOTE          16
 
 // Function-call defines ...
-#define ALLOC(_x, _n_elem)                                        \
-   do {                                                           \
-      char         **p_       = reinterpret_cast<char **>(&_x);   \
-      const size_t   n_el_    = _n_elem;                          \
-      const size_t   el_size_ = sizeof(*_x);                      \
-      const size_t   n_bytes_ = n_el_ * el_size_;                 \
-      if (_debug) {                                               \
-         *p_ = alloc_debug(n_bytes_, __FILE__, __LINE__);         \
-      } else {                                                    \
-         *p_ = alloc(n_bytes_);                                   \
-      }                                                           \
-   } while (0)
-
-#define REALLOC(_x, _no_elem, _nn_elem)                           \
-   do {                                                           \
-      char         **p_        = reinterpret_cast<char **>(&_x);  \
-      const size_t   no_el_    = _no_elem;                        \
-      const size_t   nn_el_    = _nn_elem;                        \
-      const size_t   el_size_  = sizeof(*_x);                     \
-      const size_t   no_bytes_ = no_el_ * el_size_;               \
-      const size_t   nn_bytes_ = nn_el_ * el_size_;               \
-      *p_ = ralloc(reinterpret_cast<char *>(_x),                  \
-                   no_bytes_, nn_bytes_);                         \
-   } while (0)
-
-#define FREE(_x, _n_elem)                                         \
-   do {                                                           \
-      char         **p_        = reinterpret_cast<char **>(&_x);  \
-      const size_t   n_el_    = _n_elem;                          \
-      const size_t   el_size_  = sizeof(*_x);                     \
-      const size_t   n_bytes_ = (n_el_) * (el_size_);             \
-      frea(*p_, n_bytes_);                                        \
-      _x = 0;                                                     \
-   } while (0)
+#ifdef _DEBUG
+#define ALLOC(x,norg)    alloc (#x, (char**)&x, sizeof(*x), norg)
```

Comment:
This is linked to the implementation  above, so I will let those comments speak for this location too.  In particular, having two versions with conditional compilation is less desirable than a more complicated macro that the compiler can efficiently use.